Statistics/Histograms/Histogram.cs

```diff
@@ -481,6 +567,34 @@ public bool TestForConvergence(double upperq, double lowerq)
             }
             return _Converged;
         }
+        public bool Equals(IHistogram histogramToCompare)
+        {
+            bool convergenceCriteriaAreEqual = _ConvergenceCriteria.Equals(histogramToCompare.ConvergenceCriteria);
+            if (!convergenceCriteriaAreEqual)
+            {
+                return false;
```

Comment:
Multiple return statements. Sometimes it is hard to just have one, but there are lots in here.